compiler/src/dotty/tools/dotc/tastyreflect/ReflectionCompilerInterface.scala

```diff
@@ -1459,12 +1459,12 @@ class ReflectionCompilerInterface(val rootContext: core.Contexts.Context) extend
   def Symbol_flags(self: Symbol)(given Context): Flags = self.flags
 
   def Symbol_tree(self: Symbol)(given Context): Tree = {
-    assert(!self.is(Case, butNot = Enum | Module))
+    assert(!self.is(Case) || self.isClass)
```

Comment:
```Scala
case object A
```

will translate to a term `A` with the flag `Case`:

```Scala
package <empty> {
  final lazy module case <stable> <touched> val A: A$ = new A$()
  final module case <noinits> <touched> class A$ <method> <stable> <touched>()
     extends
   Object(), _root_.scala.Product, _root_.scala.Serializable { this: A.type =>}
}
```

The same for enums:

```Scala
enum Color {
  case Red, Green, Blue
}
```

is transalted to

```Scala
    case <stable> <static> <enum> <touched> val Red: Color =
      Color.$new(0, "Red")
    case <stable> <static> <enum> <touched> val Green: Color =
      Color.$new(1, "Green")
    case <stable> <static> <enum> <touched> val Blue: Color =
      Color.$new(2, "Blue")
```

Reply:
There is no point having those assertions anyway. In the next PR `Symbol_tree` will return binds as well.